files/authenticator.sh

```diff
@@ -4,4 +4,32 @@ set -e
 cd /etc/letsencrypt
 lexicon $LEXICON_OPTIONS $LEXICON_PROVIDER $LEXICON_PROVIDER_OPTIONS create $CERTBOT_DOMAIN TXT --name="_acme-challenge.$CERTBOT_DOMAIN." --content="$CERTBOT_VALIDATION"
 
-sleep ${LEXICON_SLEEP_TIME:-30}
+if [ -z "$LEXICON_NAMESERVER" ]; then
+	NS=$(dig +short NS $CERTBOT_DOMAIN)
+else
+	NS=$(dig +short NS $CERTBOT_DOMAIN @$LEXICON_NAMESERVER)
+fi
+
+tries=0
+while : ; do
+  tries=$((tries + 1))
+  if [ $tries -ge $LEXICON_SLEEP_MAX_RETRY ]; then
```

Comment:
Here, if  `LEXICON_SLEEP_MAX_RETRY=0`, the script will always fail, since `tries=1` when we execute the first loop.

For clarity of the behavior, let's name the variable `LEXICON_SLEEP_MAX_TRY`, with the following outcomes:
* if `0`, the functionality is disabled. There is no try, and the authenticator script will just immediately return after the API calls have been sent.
* if `1`, the functionality is enabled with only one try and no retry possible
* if `n`, the functionality is enabled with one try, and `n-1` potential retries

Also, we should put the `sleep` before checking if the TXT entries are visible. This way:
* in case of `LEXICON_SLEEP_MAX_TRY=1` we have the behavior before this PR, with the bonus of checking from the authenticator script if the TXT entries are really available instead of uselessly pass back to certbot, see the challenge failing and consume for nothing some of the thresholds values set on the ACME CA server.
* we avoid to try immediately after setting the TXT entries. Indeed, if we want to check, we except a non instant propagation of the entries. Checking immediately does make very little sense in this situation.